### ISSUE TITLE
--resources can't be parsed properly

### ISSUE THREAD

**Issue by balabanmetin** [snakemake/snakemake#1069]
**Snakemake version**

Version 6.5.1

I think this is the pull request where the problem originates:

https://github.com/snakemake/snakemake/pull/1042/files
see `resources.py`

**Describe the bug**

I am running a workflow with the following command:

`snakemake -np -c8 --resources mem_mb=16384 all`

**Logs**

It raises the following error:

> (snakemake) [balaban@exp-1-07 exercise02]$ snakemake -np --resources mem_mb=16384 -c8  all
> Traceback (most recent call last):
>   File "/home/balaban/miniconda3/envs/snakemake/bin/snakemake", line 10, in <module>
>     sys.exit(main())
>   File "/home/balaban/miniconda3/envs/snakemake/lib/python3.9/site-packages/snakemake/__init__.py", line 2400, in main
>     resources = parse_resources(args.resources)
>   File "/home/balaban/miniconda3/envs/snakemake/lib/python3.9/site-packages/snakemake/resources.py", line 86, in parse_resources
>     for res, val in resources_args.items():
> AttributeError: 'list' object has no attribute 'items'

**Additional context**

Apparently  `mem_mb=16384` cannot be parsed. Note that  in the `resources.py`, `.split("=")` is moved to a new function called `decode_arg`. That function is never called. 
I also noticed that in the same pull request, developers mentioned:

> Further, --default-resources defaults are now always used.

Does that mean that we are not expected to use `--resources` directive anymore?

Someone else facing the same issue:
[https://www.biostars.org/p/9478216/](https://www.biostars.org/p/9478216)

**Comment by balabanmetin** [comment]
Minimal example 

```
# vim: set ft=python
rule all:
        input: "text.txt"

rule echo:
        output: "{sample}.txt"
        shell:
                """
                echo 'Hi!' >  {output}
                """
```


`snakemake -c1` runs normally.

`snakemake -c1 --resources mem_mb=16384` raises the error given in the previous post.

**Comment by ATpoint** [comment]
There is also a biostars post from today with a MWE and the same error:
https://www.biostars.org/p/9478216/#9478258

**Comment by johanneskoester** [comment]
Fixed by PR #1077.

**Comment by corneliusroemer** [comment]
Actually, a minimal example is
```
snakemake -c1 --resources mem_mb=16384
```

No need for a Snakefile at all!

How come this was not caught by a test? Are there no tests for CLI flags?
